test/run_test.py

```diff
@@ -1257,6 +1255,12 @@ def parse_args():
         action="store_false",
         help="Run tests without translation validation.",
     )
+    parser.add_argument(
```

Comment:
This won't work as expected: `common_util` parses the [cmdline arguments on import](https://github.com/pytorch/pytorch/blob/f9de510121be4e9e0f8fd95ed99dd5d877ac8aed/torch/testing/_internal/common_utils.py#L756-L808) and sets variables accordingly.

Some are used by `get_report_path` which would be broken afterwards:

https://github.com/pytorch/pytorch/blob/f9de510121be4e9e0f8fd95ed99dd5d877ac8aed/torch/testing/_internal/common_utils.py#L808

https://github.com/pytorch/pytorch/blob/f9de510121be4e9e0f8fd95ed99dd5d877ac8aed/torch/testing/_internal/common_utils.py#L970

I'm a bit confused about the [`run_test` function in common_utils](https://github.com/pytorch/pytorch/blob/f9de510121be4e9e0f8fd95ed99dd5d877ac8aed/torch/testing/_internal/common_utils.py#L1019) and how/where it is used but that would already pass the `--save-xml` flag on:

https://github.com/pytorch/pytorch/blob/f9de510121be4e9e0f8fd95ed99dd5d877ac8aed/torch/testing/_internal/common_utils.py#L1058-L1059

Maybe the 2 functions (in run_test and common_utils) can be merged?

Reply:
Renamed to generate xml to prevent clashing, I'll probably get rid of or change the save xml option in common_utils once I figure out what parts of CI it's affecting

Reply:
Removing the `parents=[common_parser],` means that `--save-xml` is no longer recognized and will throw an error when used. It likely affects much more from that file.
Possibly it is easier just to change https://github.com/pytorch/pytorch/pull/126690/files#diff-e72503c9e3e8766e2d1bacf3fad7b88aa166e0e90a7e103e7df99357a35df8d7R992 to `if IS_CI or options.save_xml:` and make sure `--save-xml` is passed to all subprocesses.